File: 参考/comp9041-Q/cs9041/test7/devowel_file.pl
#!/usr/bin/perl -w
$file = shift @ARGV;
open F,'<',$file or die;
@lines = <F>;
close F;

open F,'>',$file or die;
foreach $line (@lines){
  $line =~ s/[aeiou]//gi;
  print F $line;
}
close F;



















# $file = shift @ARGV;
# open F, '<',$file;
# while ($line = <F>) {
#   push @arr, $line;
# }
# close F;
# # print join "|",@arr;
# # print "\n";
# open F , '>',$file;
# print F '';
# close F;
# open F , '>>',$file;
# foreach $line (@arr){
#   $line =~ s/[aeiou]//gi;
#   print F $line;
# }

File: others/comp9041-Q/cs9041/test7/devowel_file.py
#!/usr/bin/python3
import sys, re
filename = sys.argv[1]
with open(filename) as f:
    l = [re.sub("[aeiou]","",line,flags=re.I) for line in f.readlines()]
with open(filename,'w') as f:
    for line in l:
        f.write(line)




















# import sys,re
# filename = sys.argv[1]
# with open (filename) as f:
#     l = f.readlines()
# for i in range(len(l)):
#     l[i] = re.sub('[aeiou]',"",l[i],flags=re.I)
# with open (filename,'w') as f:
#     for line in l:
#         f.write(line)
